core/domain/user_services_test.py

```diff
@@ -965,17 +966,19 @@ def test_legacy_user(self):
         last-login check was introduced.
         """
         # Set up a 'previous-generation' user.
```

Comment:
I think "previous generation user" actually means "last_logged_in = None". Might want to keep that part of the simulation if needed? Or if it's already None to begin with, assert it at the start of the test and drop this comment.

Reply:
Fixed